pkg/registry/thirdpartyresourcedata/strategy.go

```diff
@@ -75,18 +75,21 @@ func (strategy) AllowUnconditionalUpdate() bool {
 }
 
 // Matcher returns a generic matcher for a given label and field selector.
-func Matcher(label labels.Selector, field fields.Selector) generic.Matcher {
-	return generic.MatcherFunc(func(obj runtime.Object) (bool, error) {
-		sa, ok := obj.(*extensions.ThirdPartyResourceData)
-		if !ok {
-			return false, fmt.Errorf("not a ThirdPartyResourceData")
-		}
-		fields := SelectableFields(sa)
-		return label.Matches(labels.Set(sa.Labels)) && field.Matches(fields), nil
-	})
+func Matcher(label labels.Selector, field fields.Selector) *generic.SelectionPredicate {
+	return &generic.SelectionPredicate{
+		Label: label,
+		Field: field,
+		GetAttrs: func(obj runtime.Object) (labels.Set, fields.Set, error) {
+			tprd, ok := obj.(*extensions.ThirdPartyResourceData)
+			if !ok {
+				return nil, nil, fmt.Errorf("not a ThirdPartyResourceData")
+			}
+			return labels.Set(tprd.Labels), SelectableFields(tprd), nil
+		},
+	}
 }
 
-// SelectableFields returns a label set that can be used for filter selection
-func SelectableFields(obj *extensions.ThirdPartyResourceData) labels.Set {
-	return labels.Set{}
+// SelectableFields returns a field set that can be used for filter selection
+func SelectableFields(obj *extensions.ThirdPartyResourceData) fields.Set {
+	return fields.Set{}
```

Comment:
It would be more correct to return nil here to avoid an allocation.  Selectable fields is read only, correct?